pkg/remediator/watch/filteredwatcher.go

```diff
@@ -504,14 +450,48 @@ func (w *filteredWatcher) shouldProcess(object client.Object) bool {
 	// its declaration. Otherwise we expect to get another event for the same
 	// object but with a matching GVK so we can actually compare it to its
 	// declaration.
-	if object.GetObjectKind().GroupVersionKind() != decl.GroupVersionKind() {
+	currentGVK := object.GetObjectKind().GroupVersionKind()
+	declaredGVK := decl.GroupVersionKind()
+	if currentGVK != declaredGVK {
+		klog.V(5).Infof("Received a watch event for object %q with kind %s, which does not match the declared kind %s. ",
+			id, currentGVK, declaredGVK)
 		return false
 	}
 
-	if !diff.CanManage(w.scope, w.syncName, object, diff.OperationManage) {
-		w.SetManagementConflict(object, commit)
+	if diff.CanManage(w.scope, w.syncName, object, diff.OperationManage) {
+		// TODO: Remove conflict error AFTER it has been resolved, not before.
+		w.conflictHandler.RemoveConflictError(id)
+		return true
+	}
+
+	desiredManager := declared.ResourceManager(w.scope, w.syncName)
+	currentManager, found := object.GetAnnotations()[metadata.ResourceManagerKey]
+	if !found || currentManager == "" {
+		// Should never happen, since CanManage returned false.
+		klog.Errorf("Reconciler %q received a watch event for object %q, which has no manager annotation, but CanManage returned false.",
+			desiredManager, id)
+		return false
+	}
+
+	// A root reconciler can override resource managed by namespace reconciler.
+	// It is not a conflict from the root reconciler's perspective.
+	// The namespace reconciler should detect and report the conflict.
+	if w.scope == declared.RootScope && !declared.IsRootManager(currentManager) {
```

Comment:
Are we going to run into this condition? If the condition is true, `diff.CanManage()` should have already removed the conflict and returned true.

Reply:
see my other thread with sam

Reply:
Removed